### PR TITLE
hide very old images

### PR DESCRIPTION
**Pull request by monisoi**
This change is for issue https://github.com/dai-shi/remote-faces/issues/35.
Hides obsolete images that are more than 10 minutes old from receipt.

Preview URL: https://codesandbox.io/s/github/monisoi/remote-faces/tree/feat/hide-very-old-images/web/

### CODE REVIEW

**Comment by dai-shi** [comment]
Thanks for your contribution again!
I tried the Preview URL, created three peers, closed one of them, and waited more than 10min. Unfortunately, it didn't hide old images.
Looking at the code, `changed = true` is missing.

That said, as I realized it has potentially a memory leak issue, it would be better to remove the old images from the array. We do not need `tooOld` flag in this case. Do you have any concerns with it?

**Comment by monisoi** [comment]
Oh, I'm sorry!
As you pointed out, I'll try to deal with deleting the old images.

File: web/src/hooks/useFaceImages.ts [hunk 1]
```diff
@@ -75,15 +75,23 @@ export const useFaceImages = (
...
+              changed = true;
+              return { ...item, obsoleted: true };
+            }
+            if (item.received < tenMinAgo && item.obsoleted) {
+              changed = true;
+              return { ...item, userId: "tooOld" };
+            }
+            return item;
+          })
+          .filter((item) => item.userId !== "tooOld");
```
Comment (dai-shi) [inline web/src/hooks/useFaceImages.ts:93]:
```suggestion
              return null;
            }
            return item;
          })
          .filter((item) => item);
```

Maybe, this works?

**Comment by dai-shi** [comment]
あ、タイプエラーになっちった。

File: web/src/hooks/useFaceImages.ts [hunk 2]
```diff
@@ -75,15 +75,23 @@ export const useFaceImages = (
...
+              changed = true;
+              return { ...item, obsoleted: true };
+            }
+            if (item.received < tenMinAgo && item.obsoleted) {
+              changed = true;
+              return null;
+            }
+            return item;
+          })
+          .filter((item) => item);
```
Comment (dai-shi) [inline web/src/hooks/useFaceImages.ts:93]:
```suggestion
          .filter((item) => item) as typeof prev;
```
It's unfortunate that typescript can't infer this.

**Comment by dai-shi** [comment]
Merged! Thank you!

**Pull request merged**